src/Agent.Worker/Release/ArtifactProvider.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+using Agent.Worker.Release.Artifacts.Definition;
+
+using Microsoft.VisualStudio.Agent.Worker.Release.Artifacts;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Agent.Worker.Release.Artifacts;
+using Microsoft.VisualStudio.Services.ReleaseManagement.WebApi.Contracts;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Release
```

Comment:
Or just use the IExtensionManager and you don't need the IArtifactProviderManager. Take a look at this code from JobRunner.cs
```
var extensionManager = HostContext.GetService<IExtensionManager>();
IJobExtension[] extensions =
    (extensionManager.GetExtensions<IJobExtension>() ?? new List<IJobExtension>())
    .Where(x => string.Equals(x.HostType, hostType, StringComparison.OrdinalIgnoreCase))
    .ToArray();
```

Another example is how BuildJobExtension.cs uses the extension manager to finds the matching source provider.